cips/cip-3.md

```diff
@@ -0,0 +1,66 @@
+---
+cip: 3
+title: Diamonds Paid In CLOUT
+author: RedPartyHat <@redpartyhat>
+discussions-url: https://github.com/bitclout/cips/discussions/26
+created: 2021-08-04
+---
+
+## One-Line Description
+
+Pay for diamonds with CLOUT instead of creator coins.
+
+## Explanation and Motivation
+
+"Diamonds" are a tool that allow users to show their love for a creator's work on BitClout. A user
+can give up to six diamonds per post and each diamond given comes with a reward for the creator. A
+single diamond gives the creator ~0.005 USD (half a penny) at the time of writing. Each additional
+diamond 10x's the reward, up to ~$500 USD for six diamonds. This has already become a significant
+source of income for many creators.
+
+Currently, the reward given with each diamond is paid in the giver's creator coin. This is great
+for creating a dense creator coin ownership graph. Diamond recipient's often do not sell the creator
+coins that they receive, which gives them a growing financial stake in the success of all of their
+supporters. However, this is not great for compensating creators and lines diamond givers up for a
+dump event if creators need to cash out for raw CLOUT in order to support themselves.
+
+Making diamonds paid in CLOUT instead of creator coins will maximize the financial rewards for
+creators and simplify the platform for newcomers.
+
+## Specification
+
+Implementation will be similar to any forking change. A blockheight will be used to give nodes all
+nodes in the network time to upgrade. After that blockheight, all diamond transactions will be paid
+with CLOUT instead of creator coin.
+
+## Data Storage Change
+
+This change will increase the storage requirements for diamonds on the chain. Creator coins are
+stored using a balance model on the chain. Therefore, in the existing state, if a diamond receiver
+already holds the creator coin of the diamond giver the database will only be updated. In contrast,
+CLOUT is paid out using UTXOs. Each new diamond transaction will create an entry in the database
+everytime a user is given a diamond. This will be a manageable and temporary increase in storage
+that will be relieved when CLOUT is moved from UTXOs to a balance model.
```

Comment:
Very interesting. I didn't think about the fact that this will bloat UTXOs but it's a good point. It means really big creators will need to sweep periodically to avoid the app feeling slow. But even then it's not a big deal because we'll move to balance model shortly.

Reply:
I have a bunch of questions about the move to balance model - so posted it here: https://github.com/bitclout/cips/discussions/113